### ISSUE TITLE
Idea for improving documentation: Search

### ISSUE THREAD

**Issue by erhlee-bird** [hashicorp/nomad#6999]
**tl;dr** - Across the various Hashicorp products, it would be great to have an easy, empowering search experience for the documentation websites.

I was helping a coworker get up to speed with some Nomad concepts and our Nomad deployment.
He commented that the docs were really great if you knew exactly the nouns and verbs to look for.

This occurred when trying to point my coworker to documentation about the [resources stanza](https://www.nomadproject.io/docs/job-specification/resources.html). 

With my head knowledge, I know that the resources behave differently per task driver (e.g. memory upper-bound for Docker but not exec, cpu rate-limiting only when under system contention, etc). This is not reflected when looking at the documentation for the resources stanza at the specification level.

This might be solved by adding links to improve connectivity and discoverability or filling in more of the documentation. However, I think a means by which to search the documentation inline might be helpful, especially for guiding new users.

---

As an example of a site that provides the kind of experience I'm trying to describe, the documentation for [NATS](https://docs.nats.io) has a great search sidebar (powered by Algolia maybe?) that's very responsive and helps for tying together nuanced concepts from different protocols and tools in their ecosystem.

![image](https://user-images.githubusercontent.com/633268/73226758-2b189d80-413f-11ea-8c93-1ab15fdef8c5.png)

**Comment by tgross** [comment]
Hi @erhlee-bird! You'll be excited to hear that we're currently working on a revamp of the documentation sites that will include search. Stay tuned!

**Comment by tgross** [comment]
Doing some issue cleanup. Shipped in https://github.com/hashicorp/nomad/pull/8592 and improved upon in a couple PRs since then.
